askar-storage/src/backend/postgres/mod.rs

```diff
@@ -49,20 +49,58 @@ const COUNT_QUERY: &str = "SELECT COUNT(*) FROM items i
     AND (expiry IS NULL OR expiry > CURRENT_TIMESTAMP)";
 const DELETE_QUERY: &str = "DELETE FROM items
     WHERE profile_id = $1 AND kind = $2 AND category = $3 AND name = $4";
-const FETCH_QUERY: &str = "SELECT id, value,
-    (SELECT ARRAY_TO_STRING(ARRAY_AGG(it.plaintext || ':'
-        || ENCODE(it.name, 'hex') || ':' || ENCODE(it.value, 'hex')), ',')
-        FROM items_tags it WHERE it.item_id = i.id) tags
-    FROM items i
-    WHERE profile_id = $1 AND kind = $2 AND category = $3 AND name = $4
-    AND (expiry IS NULL OR expiry > CURRENT_TIMESTAMP)";
-const FETCH_QUERY_UPDATE: &str = "SELECT id, value,
-    (SELECT ARRAY_TO_STRING(ARRAY_AGG(it.plaintext || ':'
-        || ENCODE(it.name, 'hex') || ':' || ENCODE(it.value, 'hex')), ',')
-        FROM items_tags it WHERE it.item_id = i.id) tags
-    FROM items i
-    WHERE profile_id = $1 AND kind = $2 AND category = $3 AND name = $4
-    AND (expiry IS NULL OR expiry > CURRENT_TIMESTAMP) FOR NO KEY UPDATE";
+const FETCH_QUERY: &str = "
+    SELECT
+        i.id,
+        i.value,
+        ARRAY_TO_STRING(
+            ARRAY_AGG(
+                it.plaintext || ':' || ENCODE(it.name, 'hex') || ':' || ENCODE(it.value, 'hex')
+            ),
+            ','
+        ) as tags
+    FROM
+        items i
+    LEFT JOIN
+        items_tags it
+        ON it.item_id = i.id
+    WHERE
+        i.profile_id = $1
+        AND i.kind = $2
+        AND i.category = $3
+        AND i.name = $4
+        AND (i.expiry IS NULL OR i.expiry > CURRENT_TIMESTAMP)
+    GROUP BY
+    i.id,
+    i.value";
+const FETCH_QUERY_UPDATE: &str = "
+    WITH fetched_data AS (
+        SELECT
+            i.id,
+            i.value,
+            ARRAY_TO_STRING(
+                ARRAY_AGG(it.plaintext || ':' || ENCODE(it.name, 'hex') || ':' || ENCODE(it.value, 'hex')), ','
+            ) as tags
+        FROM
+            items i
+        LEFT JOIN
+            items_tags it
+            ON it.item_id = i.id
+        WHERE
+            i.profile_id = $1
+            AND i.kind = $2
+            AND i.category = $3
+            AND i.name = $4
+            AND (i.expiry IS NULL OR i.expiry > CURRENT_TIMESTAMP)
+        GROUP BY
+            i.id,
+            i.value
+    )
+    SELECT
+        *
+    FROM
+        fetched_data
+    FOR UPDATE";
```

Comment:
Any reason this isn't using `FOR NO KEY UPDATE` now?

Reply:
Great catch, thank you!

I didn't realize I had missed that.